### PR TITLE
[Testcase Refactoring] Split TestIndexingSimplification into Generic, Accelerator, and CPU classes

### PR DESCRIPTION
**Pull request by leewayx**
### Summary

Split the mixed-classification class `TestIndexingSimplification` into three classes: `TestIndexingSimplificationGeneric` (15 pure-sympy methods), `TestIndexingSimplificationAccelerator` (3 GPU methods), and `TestIndexingSimplificationCpu` (1 CPU method). Added `hw_classification` to all remaining GENERIC classes. Migrated `ReductionInvariantIndexingTests` to CUDA-only `instantiate_device_type_tests`.

### Changes

- Split `TestIndexingSimplification` into Generic/Accelerator/Cpu classes with corresponding `hw_classification` labels and instantiate calls
- Migrated `@skipUnless(HAS_GPU)` + `GPU_TYPE` to `device` parameter + `@unittest.skipIf(not HAS_TRITON)` on ACCELERATOR methods
- Migrated `@skipUnless(HAS_CPU)` to `only_for="cpu"` on CPU class
- Migrated `ReductionInvariantIndexingTests` from `@instantiate_parametrized_tests` + `HAS_CUDA_AND_TRITON` to `instantiate_device_type_tests(only_for="cuda")` + `@unittest.skipIf(not HAS_TRITON)`
- Added `hw_classification = HardwareClassification.GENERIC` to all remaining test classes and removed unused `GPU_TYPE`/`HAS_CPU`/`HAS_GPU` imports
- Removed `if HAS_CPU or HAS_GPU` guard in `__main__` block, replaced with `HAS_CPU or HAS_TRITON`

### Motivation

The original class mixed GENERIC (pure sympy), ACCELERATOR (GPU), and CPU methods under a single class without `hw_classification`, using hardcoded `GPU_TYPE` and `HAS_GPU`/`HAS_CPU` guards that excluded accelerator backends. The split enables proper device coverage via `instantiate_device_type_tests`.

### Test Plan
```
python test/inductor/test_indexing.py
```

### CODE REVIEW

File: test/inductor/test_indexing.py [hunk 1]
```diff
@@ -531,26 +538,29 @@ def foo(arg0, arg1, arg2, arg3, arg4, sentinel):
...
         compiled_foo = torch.compile(foo, fullgraph=True, dynamic=True)
         out_compiled = compiled_foo(arg0, arg1, arg2, arg3, arg4, sentinel)
         out_compiled.sum().backward()
 
-    @unittest.skipUnless(HAS_CPU, "requires CPU")
+
+class TestIndexingSimplificationCpu(InductorTestCase):
+    hw_classification = HardwareClassification.CPU
+
     def test_bool_minimum_maximum_index_propagation(self):
```
Comment (wyb2939) [inline test/inductor/test_indexing.py]: After the instantiate_device_type_tests function is used for a class, the class does not contain the device parameter in its functions.

File: test/inductor/test_indexing.py [hunk 2]
```diff
@@ -1775,8 +1808,17 @@ def test_identity_wrapped_expr_resolves_to_int(self):
...
+instantiate_device_type_tests(
+    ReductionInvariantIndexingTests, globals(), except_for="cpu", allow_xpu=True
+)
+
 if __name__ == "__main__":
     from torch._inductor.test_case import run_tests
+    from torch.utils._triton import has_triton
 
-    if HAS_CPU or HAS_GPU:
+    if HAS_CPU or has_triton():
```
Comment (wyb2939) [inline test/inductor/test_indexing.py]: After registering the capability, should the has_triton here still be retained?

**Review by FuDdd — APPROVED** [review]
